libraries/AP_Motors/AP_MotorsMulticopter.cpp

```diff
@@ -338,20 +338,37 @@ float AP_MotorsMulticopter::get_current_limit_max_throttle()
 // apply_thrust_curve_and_volt_scaling - returns throttle in the range 0 ~ 1
 float AP_MotorsMulticopter::apply_thrust_curve_and_volt_scaling(float thrust) const
 {
-    float throttle_ratio = thrust;
+    float battery_scale = 1.0;
+    if (is_positive(_batt_voltage_filt.get())) {
+        battery_scale = 1.0 / _batt_voltage_filt.get();
+    }
     // apply thrust curve - domain -1.0 to 1.0, range -1.0 to 1.0
     float thrust_curve_expo = constrain_float(_thrust_curve_expo, -1.0f, 1.0f);
     if (is_zero(thrust_curve_expo)) {
         // zero expo means linear, avoid floating point exception for small values
-        return _lift_max * thrust;
+        return _lift_max * thrust * battery_scale;
     }
+    float throttle_ratio = ((thrust_curve_expo - 1.0f) + safe_sqrt((1.0f - thrust_curve_expo) * (1.0f - thrust_curve_expo) + 4.0f * thrust_curve_expo * _lift_max * thrust)) / (2.0f * thrust_curve_expo);
+    return constrain_float(throttle_ratio * battery_scale, 0.0f, 1.0f);
+}
+
+// inverse of above
+float AP_MotorsMulticopter::remove_thrust_curve_and_volt_scaling(float throttle) const
+{
+    float battery_scale = 1.0;
     if (is_positive(_batt_voltage_filt.get())) {
-        throttle_ratio = ((thrust_curve_expo - 1.0f) + safe_sqrt((1.0f - thrust_curve_expo) * (1.0f - thrust_curve_expo) + 4.0f * thrust_curve_expo * _lift_max * thrust)) / (2.0f * thrust_curve_expo * _batt_voltage_filt.get());
-    } else {
-        throttle_ratio = ((thrust_curve_expo - 1.0f) + safe_sqrt((1.0f - thrust_curve_expo) * (1.0f - thrust_curve_expo) + 4.0f * thrust_curve_expo * _lift_max * thrust)) / (2.0f * thrust_curve_expo);
+        battery_scale = 1.0 / _batt_voltage_filt.get();
     }
-
-    return constrain_float(throttle_ratio, 0.0f, 1.0f);
+    // apply thrust curve - domain -1.0 to 1.0, range -1.0 to 1.0
+    float thrust_curve_expo = constrain_float(_thrust_curve_expo, -1.0f, 1.0f);
+    if (is_zero(thrust_curve_expo)) {
+        // zero expo means linear, avoid floating point exception for small values
+        return  throttle / (_lift_max * battery_scale);
+    }
+    float thrust = ((throttle / battery_scale) * (2.0f * thrust_curve_expo)) - (thrust_curve_expo - 1.0f);
+    thrust = (thrust * thrust) - ((1.0f - thrust_curve_expo) * (1.0f - thrust_curve_expo));
+    thrust /=  4.0f * thrust_curve_expo * _lift_max;
+    return constrain_float(thrust, 0.0f, 1.0f);
```

Comment:
This function should be able to be represented something like this:
Thrust = ((1.0 - thrust_curve_expo)  *  (throttle * _batt_voltage_filt) + thrust_curve_expo * (throttle * _batt_voltage_filt)^ 2)) / _lift_max

But reading through the work I did here I think I could restructure it to make it clearer how the expo works and why the expo works.

If you can add the automatic checking test to make sure the reverse equation works against compared to the forward equation I think I can follow this up with a PR to document and clarify how this works a little better.